controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -113,6 +116,17 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 
 	consumerConfigMapName := fmt.Sprintf("storageconsumer-%v", util.FnvHash(storageConsumer.Name))
 
+	clusterID := util.GetClusterID(ctx, r.Client, &log)
+	if clusterID == "" {
+		return reconcile.Result{}, fmt.Errorf("failed to get openshift cluster ID")
+	}
+
+	internalConsumer := clusterID == storageConsumer.Status.Client.ClusterID && storageConsumer.Status.Client.Name == storageCluster.Name
+
+	if internalConsumer {
+		consumerConfigMapName = defaults.LocalStorageConsumerConfigMapName
+	}
```

Comment:
Don't override an existing value, make a decision in an `if - else` manner when the name need to be selected